pkg/provider/pulumi/generator.go

```diff
@@ -38,10 +38,15 @@ import (
 	"github.com/nitrictech/cli/pkg/utils"
 )
 
+type PulumiOpts struct {
```

Comment:
I think it's a bit confusing calling this "Pulumi"opts as we could have a provider that doesn't use pulumi and then what do we do with it? How about StackOpts / StackActionOpts / StackDeployOpts ?
At first read, i thought it was a pulumi type..

Reply:
then you can move this type up to the top level ..
I think it's better to have generic opts rather than specialized ones for each provider type..